### PR TITLE
Add support for built-in Matrix functions

### PR DESCRIPTION
**Pull request by jiradeto**
Related issue: #465

Since all matrix built-in functions are already implemented in GraphicsFuzz, this PR mainly focuses on refactoring and documentation.

### CODE REVIEW

File: ast/src/main/java/com/graphicsfuzz/common/glslversion/Glsl330.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public String getVersionString() {
   public boolean supportedDeterminant() {
     // According to this page:
     // https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/determinant.xhtml
-    // determinant should be supported from GLSL 1.40, but glslangValidator disagrees.
```
Comment (afd) [inline ast/src/main/java/com/graphicsfuzz/common/glslversion/Glsl330.java]: Can you double-check this?  I think glslangValidator does allow it from version 1.50, so I probably just made an error.  Can you please use glslangValidator and also refer to the shading language versions to check what is the right thing to do here?

File: ast/src/main/java/com/graphicsfuzz/common/glslversion/ShadingLanguageVersion.java [hunk 2]
```diff
@@ -172,6 +172,15 @@ static ShadingLanguageVersion webGlFromVersionString(String versionString) {
    */
   boolean supportedDerivativeFunctions();
 
+  /**
```
Comment (afd) [inline ast/src/main/java/com/graphicsfuzz/common/glslversion/ShadingLanguageVersion.java]: As above, I think perhaps this comment can be revised or removed.

File: ast/src/main/java/com/graphicsfuzz/common/glslversion/ShadingLanguageVersion.java [hunk 3]
```diff
@@ -215,6 +224,15 @@ static ShadingLanguageVersion webGlFromVersionString(String versionString) {
    */
   boolean supportedInterpolationFunctions();
 
+  /**
```
Comment (afd) [inline ast/src/main/java/com/graphicsfuzz/common/glslversion/ShadingLanguageVersion.java]: Seems that inverse is supported by glslangValidator in GLSL 1.5.

File: ast/src/main/java/com/graphicsfuzz/common/glslversion/Glsl330.java (hunk 1 above)
Comment (jiradeto) [reply ast/src/main/java/com/graphicsfuzz/common/glslversion/Glsl330.java]: You are right, glslangValidator now does allow this built-in function from version 1.50.

File: ast/src/main/java/com/graphicsfuzz/common/glslversion/ShadingLanguageVersion.java [hunk 4]
```diff
@@ -234,9 +246,15 @@ static ShadingLanguageVersion webGlFromVersionString(String versionString) {
   boolean supportedMixFloatBool();
 
   boolean supportedMixNonfloatBool();
-
```
Comment (afd) [inline ast/src/main/java/com/graphicsfuzz/common/glslversion/ShadingLanguageVersion.java]: Please avoid adding whitespace like on the following line.

**Pull request merged**